surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -44,18 +48,29 @@ TestsToRun apply(Configuration configuration) {
                 + "For details on how to configure it head over to http://bit.ly/st-config");
             return Collections.emptySet();
         }
-        final Set<TestSelection> selectedTests = new LinkedHashSet<>();
+        final List<TestSelection> selectedTests = new ArrayList<>();
```

Comment:
yes, but to get all strategies list, we have to use i as `List`, otherwise we will miss strategies for other `testClasses`, as they are not adding if we are using set.

We are converting entire List into Set at the end of this function